### ISSUE TITLE
Container `#[serde(default)]` attribute ignored for `skip`ed fields

### ISSUE THREAD

**Issue by epage** [serde-rs/serde#1105]
```rust
#[derive(Debug, PartialEq, Serialize, Deserialize)]
#[serde(deny_unknown_fields, default)]
struct Skipped {
    required: Option<String>,
    #[serde(skip)] optional: String,
}

impl Default for Skipped {
    fn default() -> Self {
        Self {
            required: Some("Hello".to_owned()),
            optional: "World".to_owned(),
        }
    }
}
```
Note that `optional` is skipped and has a default value.  When deserialized to `Skipped`, `String::default()` is used instead of being filled in by `Skipped::default`, like is done when `required` is missing.

See https://play.rust-lang.org/?gist=d32b5f2e2f37bd6c8c87e403f2a26f6e&version=stable

**Comment by Binero** [comment]
Traced back the issue to this block: https://github.com/serde-rs/serde/blob/0c34e06e518c81315ec738584db0cffba6f5677a/serde_derive_internals/src/attr.rs#L884-L888

Two possible behaviours, one of which is breaking, and one of which adds some complexity to the `Field` structure. 

1. Simply require any field which has `#[serde(skip)]` set to also have either `#[serde(default)]` itself, or in its parent.
2. When the container of the `Field` has `#[serde(default)]` set, do not automatically set `#[serde(default)]` on the children.

The second option would require exposing some context to the field.

**Comment by dtolnay** [comment]
@Binero fixed this in #1115 and I released the fix in serde_derive 1.0.24. Thanks!
